File: android/DataBinding/app/src/main/java/com/aptinfo/databinding/MainActivity.java
package com.aptinfo.databinding;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;

import android.os.Bundle;

import com.aptinfo.databinding.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {
    private String text = "Hello Data Binding";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ActivityMainBinding binding = DataBindingUtil.setContentView(this, R.layout.activity_main);
        binding.setText(text);
    }
}

File: android/DataBindingKt/app/src/main/java/com/aptinfo/databindingkt/MainActivity.kt
package com.aptinfo.databindingkt

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.databinding.DataBindingUtil
import com.aptinfo.databindingkt.R
import com.aptinfo.databindingkt.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        //setContentView(R.layout.activity_main)
        val binding: ActivityMainBinding = DataBindingUtil.setContentView(this, R.layout.activity_main)

        binding.text = "Hello Data Binding"
    }
}
